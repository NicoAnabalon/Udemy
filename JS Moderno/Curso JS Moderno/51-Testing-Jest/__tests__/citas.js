import Citas from '../js/classes/Citas';

describe("Probar la clase de Citas", () => {
    const citas = new Citas();
    const id = Date.now();

    test("Agregar una nueva cita", () => {

        const citaObj = {
            mascota: 'Atenea',
            propietario: 'Nicolas',
            telefono: '+56996747517',
            fecha: '21-08-2021',
            hora:'23:30',
            sintomas: 'Es una buena perrita'
        };

        citaObj.id = id;

        citas.agregarCita(citaObj);

        // Prueba
        expect(citas).toMatchSnapshot();
    });

    test("Actualizar cita", () => {
        const citaActualizada = {
            id,
            mascota: 'Eros',
            propietario: 'Nicolas',
            telefono: '+56996747517',
            fecha: '21-08-2021',
            hora:'23:30',
            sintomas: 'Es un buen gatito'
        };

        citas.editarCita(citaActualizada);

        expect(citas).toMatchSnapshot();
    });

    test("Eliminar cita", () => {
        citas.eliminarCita(id);

        expect(citas).toMatchSnapshot();
    })
});