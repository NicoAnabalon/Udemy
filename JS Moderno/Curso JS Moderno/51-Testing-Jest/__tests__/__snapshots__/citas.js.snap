// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probar la clase de Citas Actualizar cita 1`] = `
Citas {
  "citas": Array [
    Object {
      "fecha": "21-08-2021",
      "hora": "23:30",
      "id": 1629604743498,
      "mascota": "Eros",
      "propietario": "Nicolas",
      "sintomas": "Es un buen gatito",
      "telefono": "+56996747517",
    },
  ],
}
`;

exports[`Probar la clase de Citas Agregar una nueva cita 1`] = `
Citas {
  "citas": Array [
    Object {
      "fecha": "21-08-2021",
      "hora": "23:30",
      "id": 1629604743498,
      "mascota": "Atenea",
      "propietario": "Nicolas",
      "sintomas": "Es una buena perrita",
      "telefono": "+56996747517",
    },
  ],
}
`;

exports[`Probar la clase de Citas Eliminar cita 1`] = `
Citas {
  "citas": Array [],
}
`;
